megamek/src/megamek/common/Game.java

```diff
@@ -1402,7 +1406,6 @@ public synchronized void removeEntity(int id, int condition) {
             return;
         }
 
-        entities.remove(toRemove);
         entityIds.remove(Integer.valueOf(id));
```

Comment:
Shouldn't we also remove the id before the if statement?

Reply:
Probably wouldn't hurt, but, unlike entities, we don't use that lookup table to drive "get me all the entities in the game" functionality.